churchland_pipeline_python/utilities/dataviz.py

```diff
@@ -0,0 +1,443 @@
+
+"""Data visualization.
+
+Todo:
+    * overlay multiple attributes
+    * apply functions to attributes
+    * colormaps
+"""
+
+import datajoint as dj
+import inspect, re, math
+import numpy as np
+import matplotlib.pyplot as plt, matplotlib.colors as mplcol
+import colorcet as cc
+from typing import NewType, Tuple, List, Any
+
+import timeit
+
+DataJointTable = dj.user_tables.OrderedClass
+
+# -------
+# GLOBALS
+# -------
+
+# default figure settings
+DEFAULT = dict(
+    layout = dict(
+        figsize=(12,8),
+        n_rows=None,
+        n_columns=None,
+        grid_attr=(),
+        orientation=None,
+        limit_figures=10,
+        limit_subplots=25,
+        limit_layers=500,
+        limit_rows=50,
+        limit_columns=10,
+    ),
+    style = dict(
+        kind='line',
+        marker_size=3,
+        color=None,
+        color_map=cc.blues,
+        reverse_color_map=False,
+        remove_spines=('top', 'right'),
+        show_x_ticks=True,
+        show_y_ticks=True,
+    ),
+    axes = dict(
+        sharex=False,
+        sharey=False,
+        x_lim=(None, None),
+        y_lim=(None, None),
+        y_tick_step=None,
+        show_legend=False,
+        sparse_legend=False,
+    )
+)
+
+
+# ----------
+# PLOT TABLE
+# ----------
+
+def plot_table(
+    table: DataJointTable, 
+    y: str,
+    x: str=None,
+    group_by: Tuple[str]=(),
+    stack_by: Tuple[str]=(),
+    ignore: Tuple[str]=(),
+    track: Tuple[str]=(),
+    map_fcn: Any=None,
+    map_args: Any=None,
+    layout: dict=None,
+    style: dict=None,
+    axes: dict=None,
+    labels: dict=None,
+    ) -> None:
+    """Plot DataJoint table.
+
```

Comment:
One suggestion is to put a check of how many data points to plot before actually plot it. If too many, it might not be renderable.